thirdparty/kyaml/runfn/runfn.go

```diff
@@ -152,37 +132,92 @@ func (r RunFns) getNodesAndFilters() (
 		return nil, nil, outputPkg, err
 	}
 
-	fltrs, err := r.getFilters(buff.Nodes)
+	fltrs, err := r.getFilters()
 	if err != nil {
 		return nil, nil, outputPkg, err
 	}
+
+	// we need to exclude function config file if we don't want to include
+	// meta resources.
+	if !r.IncludeMetaResources {
+		p := kio.Pipeline{
+			Inputs:                []kio.Reader{buff},
+			Filters:               []kio.Filter{kio.FilterFunc(r.filterOutFnConfigFile)},
+			Outputs:               []kio.Writer{buff},
+			ContinueOnEmptyResult: r.ContinueOnEmptyResult,
+		}
+		if err := p.Execute(); err != nil {
+			return nil, nil, outputPkg, err
+		}
+	}
 	return buff, fltrs, outputPkg, nil
 }
 
-func (r RunFns) getFilters(nodes []*yaml.RNode) ([]kio.Filter, error) {
-	var fltrs []kio.Filter
-
-	// fns from annotations on the input resources
-	f, err := r.getFunctionsFromInput(nodes)
-	if err != nil {
-		return nil, err
+// filterOutFnConfigFile is a filter which filters out the function config
+// file
+func (r RunFns) filterOutFnConfigFile(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
```

Comment:
Can we rename `nodes` to `input` and `result` to `output` ?

Reply:
done